sorted_list_dict.py

```diff
@@ -12,20 +12,36 @@ def __eq__(self, other):
     def __str__(self) -> str:
         return self.__list.__str__()
 
+    def __len__(self) -> int:
+        return self.__list.length()
+
     def __find_gte_or_length(self, key: str) -> int:
         '''
         gte means greater or equal
         '''
-        # TODO: binary search
+        # binary search
```

Comment:
This is how I implemented binary search. It seems there are better ways, but I was able to do it this way.